tests/test_logging.py

```diff
@@ -86,7 +86,9 @@ def get(self):
         # debug log shown from ns1
         client.get("/ns1/")
         matching = [r for r in caplog.records if r.message == "hello from ns1"]
-        assert len(matching) == 1
+        # NOTE: previously set to 1; 0 is accurate as debug level is lower than
+        # info level, which is set to the root logger
+        assert len(matching) == 0
```

Comment:
This test is (for some reason) related to `bleach` dependency (as changing it's version, to support higher pyhton version, made this test fail).

Reply:
did you try printing the output before your changes and after to see what differs? this looks like a domain name server connectivity check

Reply:
Updated this test.

The issue was pytest's `LogCaptureFixture` and the fact that when settings the level to a logger trough an instance of that class it would update the log level of that instance's handler (in charge of intercepting messages for assertions). In turn, trying to send a debug log when the level is info would fail.